src/WatchDogs.Application/Program.cs

```diff
@@ -42,14 +48,16 @@
     });
 
     builder.Services.AddSingleton<IDxTradeAuthenticator, DxTradeAuthenticator>();
-    builder.Services.AddSingleton<ISessionTokenStorage, InMemorySessionTokenStorage>(); 
+    builder.Services.AddSingleton<ISessionTokenStorage, InMemorySessionTokenStorage>();
 
     builder.Services.AddSingleton<DxTradeClient>();
+    builder.Services.AddTransient<DataInserter>();
     builder.Services.AddTransient<IFakeTradeGenerator, FakeTradeGenerator>();
-    builder.Services.AddSingleton(serviceProvider =>
+    builder.Services.AddTransient(serviceProvider =>
     {
         var dataGenerator = serviceProvider.GetRequiredService<IFakeTradeGenerator>();
-        return new Watcher(TimeSpan.FromMilliseconds(1000), dataGenerator);
+        var dataInserter = serviceProvider.GetRequiredService<DataInserter>();
+        return new Watcher(TimeSpan.FromMilliseconds(1000), dataGenerator, dataInserter);
```

Comment:
Tbh I would provide this value from config and inject it as an `IOptions`

Reply:
This took 30 minutes :D but finally last change is done and I am merging